### DISCUSSION TITLE
Secrets

### DISCUSSION THREAD

**Discussion by dialanothernumb** [htpcBeginner/docker-traefik#156] [Support]
When I run docker-compose I get this error

> ERROR: Service "traefik" uses an undefined secret "cloudflare_email"

I am following the [guide](https://www.smarthomebeginner.com/traefik-docker-security-best-practices/#8_Use_Docker_Secrets)
and I have created secrets files in the docker/secrets folder as root so I am assuming this is a permissions problem? Can anyone advise me how I might be going wrong?

**Comment by Liquidtouch** [comment]
Did you mirror the github repository and follow the guide? The guide is not up to date with the repository. 

Double check the .yml file has the correct path to the secret folder. When I had a similar issue it was because I created and was editing a file based on the guide but the code from repo was looking in a different place for the file and I was updating the wrong spot. In the end I wiped out my server, mirrored the repo freshly again, and now don't use the guide for step by step instructions only to get a sense of what I need to configure to get it working.
